.github/workflows/release.yml

```diff
@@ -41,32 +41,74 @@ jobs:
         with:
           command: build
           args: --target ${{ matrix.target }} --release
+
+      - name: Compress and rename executable
+        if: ${{ matrix.os != 'windows-latest' }}
+        run: gzip -c target/${{ matrix.target }}/release/svd2rust > svd2rust-${{ matrix.target }}${{ matrix.suffix }}
       - name: Rename executable
-        run: mv target/${{ matrix.target }}/release/svd2rust${{ matrix.suffix }} target/${{ matrix.target }}/release/svd2rust-${{ matrix.target }}${{ matrix.suffix }}
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: mv target/${{ matrix.target }}/release/svd2rust${{ matrix.suffix }} svd2rust-${{ matrix.target }}${{ matrix.suffix }}
+
       - uses: actions/upload-artifact@v3
         with:
           name: svd2rust-${{ matrix.target }}
-          path: target/${{ matrix.target }}/release/svd2rust-${{ matrix.target }}${{ matrix.suffix }}
+          path: svd2rust-${{ matrix.target }}${{ matrix.suffix }}
 
   release:
     name: release
     runs-on: ubuntu-latest
     needs: [build]
     steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
       - uses: actions/download-artifact@v3
         with:
           path: artifacts
       - run: ls -R ./artifacts
+
+      - name: Set current date as environment variable
+        run: echo "CURRENT_DATE=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+
+      - uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: markdown-extract
+      - name: Extract unreleased changelog
+        id: extract-unreleased-changelog
+        if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
+        run: |
+          CHANGELOG=$(markdown-extract Unreleased CHANGELOG.md)
+          LATEST_TAG=$(git tag --sort=v:refname |& tail -n 1)
+          CHANGELOG="[Latest Commit](https://github.com/rust-embedded/svd2rust/commit/${{ github.sha }})"$'\n\n'${CHANGELOG}$'\n\n'"[Unreleased]: https://github.com/rust-embedded/svd2rust/compare/${LATEST_TAG}...HEAD"
+          CHANGELOG="${CHANGELOG//'%'/'%25'}"
+          CHANGELOG="${CHANGELOG//$'\n'/'%0A'}"
+          CHANGELOG="${CHANGELOG//$'\r'/'%0D'}"
+          echo "::set-output name=changelog::$CHANGELOG"
```

Comment:
I think this overall is wrong, this should be added to the changelog from the start, and not added afterwards in the release

Reply:
I think you mean marking the `Unreleased` section as `vX.Y.Z`, building binaries, and creating release should be in one action? If so, I need to think about how to reimplement this action.

It looks like all the previous releases are done by hand, so I just add action to build binaries and upload them.

Reply:
No, this specifically I mean that the changelog should already contain this information, e.g the changeset between this release and the last, and on tag pushes the changelog should use the changelog entry for that specific version.

A release tag should not have its changelog entries under unreleased

Reply:
I see. Need time to figure out how to extract that information.

Reply:
See for example how it was solved in cross

https://github.com/cross-rs/cross/blob/4deac830a4992a70f7cb070eda38d5ff7cfcbf86/.github/actions/cargo-publish/action.yml#L11